planner/languages/java/java_planner.go

```diff
@@ -17,9 +35,68 @@ func (p *Planner) Name() string {
 }
 
 func (p *Planner) IsRelevant(srcDir string) bool {
-	return false
+	// Checking for pom.xml (maven) only for now
+	// TODO: add build.gradle file detection
+	pomXMLPath := filepath.Join(srcDir, "pom.xml")
+	return fileExists(pomXMLPath)
 }
 
 func (p *Planner) GetPlan(srcDir string) *plansdk.Plan {
-	return &plansdk.Plan{}
+	javaPkg := getJavaPackage(srcDir)
+	return &plansdk.Plan{
+		DevPackages: []string{
+			javaPkg,
+			defaultMaven,
+		},
+	}
+}
+
+func getJavaPackage(srcDir string) string {
+	pomXMLPath := filepath.Join(srcDir, "pom.xml")
+	javaVersion := parseJavaVersion(pomXMLPath)
+	v, ok := jVersionMap[javaVersion]
+	if ok {
+		return v
+	} else {
+		return defaultJava
+	}
+}
+
+func parseJavaVersion(pomXMLPath string) int {
+	parsedVersion, err := parseXML(pomXMLPath)
+	if err != nil {
+		fmt.Printf("error parsing java version from pom file: %v", err)
+		return 0
+	}
+	return parsedVersion
+}
+
+func fileExists(path string) bool {
+	_, err := os.Stat(path)
+	return err == nil
+}
+
+func parseXML(pomXMLPath string) (int, error) {
+	// read the XML file as a byte array.
+	byteArray, err := os.ReadFile(pomXMLPath)
+	if err != nil {
+		return 0, errors.WithStack(err)
+	}
+
+	var project mvnparser.MavenProject
+	// unmarshaling byteArray which contains our pom file content into 'project'
+	err = xml.Unmarshal(byteArray, &project)
+	if err != nil {
+		return 0, errors.WithStack(err)
+	}
```

Comment:
Not blocking, but once #120 lands, these lines could become:
```
var project mvnparse.MavenProject
err = cuecfg.ParseFile(pomXMLPath, &project)
```

That PR adds support for xml parsing to `cuecfg` package.

Reply:
I copied the same changes in `cuecfg` package that you made in #120 . So if either of these PRs merge not conflict will happen (hopefully)